### PR TITLE
React shared components

### PR DESCRIPTION
**Pull request by holiber**
Is safe to merge now

### CODE REVIEW

File: app/components-react/shared/inputs/ListInput.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  function onSearchHandler(searchStr: string) {
+    p.onSearch && p.onSearch(searchStr);
+  }
+
+  return render();
+});
+
+export function renderOption(opt: IListOption, ind: number, inputProps: ICustomListProps) {
```
Comment (gettinToasty) [inline app/components-react/shared/inputs/ListInput.tsx:79]: is this an accidental export?

File: app/components/windows/go-live/GoLiveError.tsx [hunk 2]
```diff
@@ -83,8 +83,6 @@ export default class GoLiveError extends TsxComponent<{}> {
         return this.renderRestreamError(error);
       case 'YOUTUBE_STREAMING_DISABLED':
         return this.renderYoutubeStreamingDisabled(error);
-      case 'YOUTUBE_PUBLISH_FAILED':
-        return this.renderYoutubePublishError(error);
```
Comment (gettinToasty) [inline app/components/windows/go-live/GoLiveError.tsx:87]: not sure why this is removed?

File: app/components-react/shared/inputs/ListInput.tsx (hunk 1 above)
Comment (gettinToasty) [reply app/components-react/shared/inputs/ListInput.tsx:79]: disregard, i see where it's used

File: app/components-react/shared/inputs/ListInput.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+// select what features from the antd lib we are going to use
+const ANT_SELECT_FEATURES = [
+  'showSearch',
+  'loading',
+  'placeholder',
+  'notFoundContent',
+  'onDropdownVisibleChange',
+  'onSearch',
+  'onSelect',
+] as const;
```
[the remark this answers is not in the document]
Comment (holiber) [reply app/components-react/shared/inputs/ListInput.tsx:17]:
```
const a = ['foo', 'bar']; // has type string[]
const b = ['foo', 'bar'] as const; // has type ['foo', 'bar']
```

File: app/components/windows/go-live/GoLiveError.tsx (hunk 2 above)
Comment (holiber) [reply app/components/windows/go-live/GoLiveError.tsx:87]: This accidentaly came from my original big PR. YT should automatically publish the broadcast now, this error is not happening anymore

**Review by gettinToasty — APPROVED** [review]

**Pull request merged**